router/rate_limiting.go

```diff
@@ -0,0 +1,109 @@
+package router
+
+import (
+	"container/list"
+	"context"
+	"sync"
+	"time"
+
+	"github.com/topfreegames/pitaya/cluster"
+	"github.com/topfreegames/pitaya/config"
+	"github.com/topfreegames/pitaya/constants"
+	"github.com/topfreegames/pitaya/errors"
+	"github.com/topfreegames/pitaya/logger"
+	"github.com/topfreegames/pitaya/route"
+	"github.com/topfreegames/pitaya/session"
+)
+
+// WithRateLimiting wraps routingFunc and controls
+// player rate through session.
+func WithRateLimiting(c *config.Config, f RoutingFunc) RoutingFunc {
```

Comment:
is it possible to use an implementation of a pitaya pipeline instead of a decorator function?